firmware/cmake/generate_cubemx.mk

```diff
@@ -34,4 +37,4 @@ Makefile: $(IOC_FILE)
 # This is used by build_cubemx.cmake
 	echo "" >> Makefile
 	echo "%.value:" >> Makefile
-	echo -e "\t@echo \$$(\$$*)" >> Makefile
\ No newline at end of file
+	echo "\t@echo \$$(\$$*)" >> Makefile
```

Comment:
I don't know why I had this

Reply:
https://www.gnu.org/software/coreutils/manual/html_node/echo-invocation.html#echo-invocation

Without `-e`, echo will print `\t` instead of a tab character

Reply:
This is the output when adding -e on my end:
```Make
%.value:
-e 	@echo $($*)
```

Without -e:
```Make
%.value:
	@echo $($*)
```

It is likely this is a unix vs windows thing.

Reply:
This line also gives me correct output, could you give it a try on your end?
```Bash
echo '\t@echo $$($$*)' >> Makefile
```

Reply:
What we really need here is to get builds working in regressions. @Micnasr do you have time to help us out with this?

Reply:
> This line also gives me correct output, could you give it a try on your end?
> 
> ```shell
> echo '\t@echo $$($$*)' >> Makefile
> ```

This does not work in git-bash. I have a PR to this branch to fix all of this

Reply:
@samparent97 I am looking at the PR and the changes but I am struggling to understand what we are trying to achieve here and what the problem is

Reply:
@Micnasr slightly unrelated to this PR. But if you could get builds running as a github action to block merge that would be great. Check out issue #69